include/vcpkg/base/stringview.h

```diff
@@ -86,3 +88,45 @@ namespace vcpkg
 
 VCPKG_FORMAT_AS(vcpkg::ZStringView, vcpkg::StringView);
 VCPKG_FORMAT_AS(vcpkg::StringLiteral, vcpkg::StringView);
+
+template<::size_t N>
+struct StringArray : public std::array<char, N>
+{
```

Comment:
```suggestion
{
    static_assert(N != 0, "Space for the null terminator is necessary.");
```

Reply:
No joke, `static_assert` is not evaluated to constexpr😂. Tgerefore, I have to use `enable_if`.

Reply:
I ended up using `std::enable_if`

Reply:
That doesn't have the desired behavior though; we want attempting to form StringArray<0> to be a hard error with a message describing why, not SFINAE out.